### PR TITLE
More lua enums

### PR DESCRIPTION
**Pull request by C-ffeeStain**
I am now done, so merge this ASAP.
I am adding more enums to the Lua API such as:
- CHAR_STATE (done)
- ENT_FLAG (done)

### CODE REVIEW

File: src/game_api/script/usertypes/char_state.cpp [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        "ENTERING",
+        19,
+        "LOADING",
+        20,
+        "EXITING",
+        21,
+        "DYING",
+        22);
+}
+} // namespace NCharacterState
```
Comment (Malacath-92) [inline src/game_api/script/usertypes/char_state.cpp:46]: Can you add new lines at the end of files please.  🙏

File: src/game_api/script/usertypes/char_state.hpp [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#pragma once
+
+#include <sol/forward.hpp>
+
+class ScriptImpl;
+
+namespace NCharacterState
+{
+void register_usertypes(sol::state& lua);
+};
```
Comment (Malacath-92) [inline src/game_api/script/usertypes/char_state.hpp:10]: Newline again  :)

File: src/game_api/script/usertypes/flags_lua.cpp [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        "PAUSE_AI_AND_PHYSICS",
+        28,
+        "DEAD",
+        29,
+        "HAS_BACKITEM",
+        32
+
+    );
+}
+}; // namespace NEntityFlags
```
Comment (Malacath-92) [inline src/game_api/script/usertypes/flags_lua.cpp:70]: Also

File: src/game_api/script/usertypes/flags_lua.hpp [hunk 4]
```diff
@@ -0,0 +1,10 @@
+#pragma once
+
+#include <sol/forward.hpp>
+
+class ScriptImpl;
+
+namespace NEntityFlags
+{
+void register_usertypes(sol::state& lua);
+};
```
Comment (Malacath-92) [inline src/game_api/script/usertypes/flags_lua.hpp:10]: here  :P

**Review by Malacath-92 — CHANGES_REQUESTED** [review]

File: src/game_api/script/usertypes/char_state.cpp (hunk 1 above)
Comment (C-ffeeStain) [reply src/game_api/script/usertypes/char_state.cpp:46]: Sure. Don't know why it's requested by most programs though.

**Review by Malacath-92 — APPROVED** [review]

**Pull request merged**